fe/fe-core/src/main/java/org/apache/doris/cloud/qe/SnapshotProxy.java

```diff
@@ -0,0 +1,92 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.cloud.qe;
+
+import org.apache.doris.cloud.proto.Cloud;
+import org.apache.doris.cloud.rpc.MetaServiceProxy;
+import org.apache.doris.common.Config;
+import org.apache.doris.rpc.RpcException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class SnapshotProxy {
```

Comment:
It is unnecessary to extract the visible version methods into SnapshotProxy. If you want to reorganize the code, then you should put this kind of helper class in the cloud.rpc package, and named it `VersionHelper` might be better.

BTW, you have put the code for getting the version in CloudPartition in SnapshotProxy, should the code for getting the version in OlapTable also be put here?

Reply:
Because table version and partition visible version use the same rpc interface, there will be a lot of duplicate code. This SnapshotProxy is designed to put duplicate code in one place.
If the name and package need to be changed, I can modify it in the next PR which will fix a regression test case.

Reply:
Hi, I finish it in #32989. Please review it.